test/extended/networking/egressqos.go

```diff
@@ -0,0 +1,594 @@
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Verifying the pods egress traffic is marked with the updated dscp values")
+			pingAndCheckDSCP(oc, f, target1, target2, dscpValue-11, dscpValue-12)
+
+			g.By("Deleting the EgressQoS")
+			_, err = runOcWithRetry(oc.AsAdmin(), "delete", "-f", yamlPath)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Verifying the pods egress traffic is not marked with DSCP")
+			pingAndCheckDSCP(oc, f, target1, target2, 0, 0)
+		})
+	})
+})
+
+func launchEgressQoSSrcPod(f *e2e.Framework, nodeName string, generateName string, labels map[string]string) (*v1.Pod, error) {
+	contName := fmt.Sprintf("%s-container", generateName)
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: generateName,
+			Labels:       labels,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    contName,
+					Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+					Command: []string{"sleep", "100000"},
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+
+	p, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.Background(), pod, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	err = e2epod.WaitForPodRunningInNamespace(f.ClientSet, p)
+	return p, err
+}
+
+func launchHostNetworkedPodForTCPDump(f *e2e.Framework, tcpdumpImage, nodeName, generateName string) (*v1.Pod, error) {
+	contName := fmt.Sprintf("%s-container", generateName)
+	runAsUser := int64(0)
+	securityContext := &v1.SecurityContext{
+		RunAsUser: &runAsUser,
+		Capabilities: &v1.Capabilities{
+			Add: []v1.Capability{
+				"SETFCAP",
+				"CAP_NET_RAW",
+				"CAP_NET_ADMIN",
+			},
+		},
+	}
+
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: generateName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:            contName,
+					Image:           tcpdumpImage,
+					Command:         []string{"sleep", "100000"},
+					SecurityContext: securityContext,
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+			HostNetwork:   true,
+		},
+	}
+
+	p, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.Background(), pod, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	err = e2epod.WaitForPodRunningInNamespace(f.ClientSet, p)
+	return p, err
+}
+
+func updatePodLabels(f *e2e.Framework, pod *v1.Pod, labels map[string]string) error {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		p, err := f.ClientSet.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		p.Labels = labels
+		_, err = f.ClientSet.CoreV1().Pods(pod.Namespace).Update(context.TODO(), p, metav1.UpdateOptions{})
+		return err
+	})
+
+	return err
+}
```

Comment:
nit: as I don't see anything QOs specific in this utility methods, could they be placed on a more reusable package?

Reply:
done, moved to `networking/util`
changed `launchEgressQoSSrcPod` -> `createSleepingPod`